lint-gitlab-ci.sh

```diff
@@ -1,27 +1,46 @@
 #!/usr/bin/env bash
 set -e
 
-GITLAB_SERVER_URL=${GITLAB_SERVER_URL:-"gitlab.com"}
+# GITLAB_SERVER_URL=${GITLAB_SERVER_URL:-"gitlab.com"}
+# ## get gitlab push remote
+GITLAB_REMOTE=$(git remote -v | grep push | awk '{print $2}')
+# ## remove useless parts
+# ## git urls (everything bevore @)
+GITLAB_REMOTE=${GITLAB_REMOTE#*@}
+# ## https urls (everythin before //)
+GITLAB_REMOTE=${GITLAB_REMOTE#*//}
+# ## .git ending
+GITLAB_REMOTE=${GITLAB_REMOTE%.git}
+
+# ## get the url
+GITLAB_SERVER_URL=$(echo "${GITLAB_REMOTE:?}" | cut -d':' -f1)
+# ## get the project name and URL encode
+GITLAB_PROJECT=$(echo "${GITLAB_REMOTE:?}" | cut -d':' -f2 | sed 's#/#%2F#g' )
+# ## get project ID
+# GITLAB_PROJECT_ID=$(curl -s --header "PRIVATE-TOKEN: ${GITLAB_TOKEN:?}" "https://${GITLAB_SERVER_URL:?}/api/v4/projects/${GITLAB_PROJECT:?}" | jq '.id')
+
 CI_FILE=${1:-".gitlab-ci.yml"}
 TOKEN=${GITLAB_TOKEN:-""}
-URL="https://${GITLAB_SERVER_URL}/api/v4/ci/lint?include_merged_yaml=true"
+# ## ENDPOINT: see https://docs.gitlab.com/ee/api/lint.html
+# ## old endpoint, deprecated since 15.7, removed in 16.0
+# API_ENDPOINT=ci/lint
+# ## new endpoint: /projects/:id/ci/lint
+API_ENDPOINT=projects/${GITLAB_PROJECT:?}/ci/lint
+URL="https://${GITLAB_SERVER_URL}/api/v4/${API_ENDPOINT:?}?include_merged_yaml=true"
```

Comment:
`URL="https://${GITLAB_SERVER_URL}/api/v4/projects/${GITLAB_PROJECT_ID:?}/ci/lint` worked for me

Reply:
Is the change I proposed using `API_ENDPOINT` based on `GITLAB_PROJECT` not working for you?

It should work as well, because as mentioned above, `:id` can be the actual ID or the URL encoded project path.